Nav buttons

rhenck opened the pull request:
Hier mal ein erster Aufschlag für die Navigationsknöpfe.
Inhaltlich denke ich selbsterklärend bzw. haben wir ja über den Plan gesprochen.

Es gibt eine neue Komponente, die man mit "nav-button-group::previous##next##first##last##end" bedienen kann. Fehlende Parameter erzeugen einen Fehler. Das Event vopUnitNavigationRequestedNotification wird geworfen.

Habe die Dev Variante und im Testbed getestet und klappt. (Wobei das Testbed damit nichts anfängt, aber das Event kommt an.)

rhenck committed:
Change build target filename depending on dev param

This makes it possible to differentiate between the output variants.

rhenck committed:
Add Material Icons module

rhenck committed:
Add Nav Button Component

rhenck committed:
Add Nav-event infrastructure

rhenck commented:
For die iqb-scripted Spezifikation kommt gleich ein PR hinterher.

mechtelm commented:
* es gibt einen Syntaxfehler in UIElements.ts: schließende Klammer fehlt vor NavButtonGroupElement
* ist der äußere Rand wieder verloren gegangen (page-div padding)
* "player-component alive!"?
* end-button: besser icon `home` oder `exit_app`?
* Leerzeile im Script wirft Fehler, wenn noch andere (unsichtbare) Whitespaces enthalten sind
* Oberer Abstand der Button-Group ist zu knapp

mechtelm commented:
Die If-Konstruktion funktioniert bei Abfrage einer Chechbox-Variablen nicht richtig. Es sollte bei 1 der True-Zweig, bei 0 der else-Zweig genommen werden.

mechtelm commented:
Das Wiederherstellen von Werten in If-Bereichen funktioniert nicht immer. Nach etwas hin und her geht schließlich die Eingabe verloren. Hier meine Spieldaten:

[abi-player-example-units.zip](https://github.com/iqb-berlin/verona-player-abi/files/6128531/abi-player-example-units.zip)

rhenck commented:
Da ging anscheinend einiges schief beim Mergen des Master-Branch. Da habe ich nicht nochmal durchgetestet und angenommen das wird schon passen. Ich schaue mir die Sachen nochmal an.

rhenck committed:
Merge branch 'master' into nav-buttons

rhenck committed:
Fix syntax error after merge

rhenck committed:
Add examples for nav buttons to dev wrapper

rhenck commented:
`end-button: besser icon home oder exit_app?`

Dann entscheide dich mal. Ich finde meinen von den dreien am besten. Wenn ich zwischen deinen beiden wählen müsste, dann finde ich home besser. Aber letztlich ist mir das herzlich egal. :)

rhenck committed:
Fix error with empty line with whitespaces

rhenck committed:
Add top/bottom margin to nav buttons

mechtelm commented:
Wenn wir streng semantisch agieren wollen und auch nach dem nächsten Material-Release noch zutreffende Icons haben wollen, dann müssten wir folgende nehmen: arrow_back, arrow_forward, first_page, last_page, and dann, tja. Also wennste meine Vorschläge nicht magst dann vielleicht arrow_upward? keyboard_capslock erinnert mich (ja auch an die Großbuchstabentaste) an das Auswerfen einer CD.

mechtelm commented:
Ähm also keyboard_arrow_right und keyboard_arrow_left sind definitiv schöner, die gehen bestimmt auch nach dem nächsten M-Release...

mechtelm commented:
keyboard_arrow_up ist auch schön

rhenck commented:
`keyboard_capslock erinnert mich (ja auch an die Großbuchstabentaste) an das Auswerfen einer CD`
Ja das stimmt. :D
Ich habe die wirklich nicht nach Semantik ausgesucht, sondern nur danach, wie sie mir gefallen und irgendwie gefällt mir das Symbol, auch wenn es nicht ganz passend ist.

Wenn wir die 'korrekten' icons nehmen wollen, dann wäre wohl arrow-upward das Richtige. Oder home_work.

Könnte so ein Pfeil nach oben auch missverstanden werden, dass man denkt der scrollt nach oben zum Anfang der Unit? Da wäre das CapsLock Icon evtl. besser, weil es deutlicher macht, dass irgendwas Beendet (oder halt Ausgeworfen) wird.

Na wie auch immer.. Also einigen wir uns auf "keyboard_arrow_up" ?

rhenck commented:
> Die If-Konstruktion funktioniert bei Abfrage einer Chechbox-Variablen nicht richtig. Es sollte bei 1 der True-Zweig, bei 0 der else-Zweig genommen werden.

Verstehe ich das richtig, wenn nichts ausgewählt ist (also der Startzustand) ist der Else-Inhalt sichtbar? Ich bin mir ziemlich sicher, da nichts geändert zu haben.

rhenck commented:
> Das Wiederherstellen von Werten in If-Bereichen funktioniert nicht immer. Nach etwas hin und her geht schließlich die Eingabe verloren. Hier meine Spieldaten:
> 
> [abi-player-example-units.zip](https://github.com/iqb-berlin/verona-player-abi/files/6128531/abi-player-example-units.zip)

Hier wäre eine etwas detailierte Beschreibung gut gewesen... :)
Ich habe gemerkt, dass er Kinder von IfBlock innerhalb von Repeatsblöcken nicht wiederherstellt. Sind sonst noch Dinge, die nicht gehen?

mechtelm commented:
Ja ok keyboard_arrow_up

mechtelm commented:
Zu if bei Checkboxes: Probier mal: Zum Start ist wird weder der true- noch der false-Zweig gezeigt, nach dem Klick nur der else-Zweig, und nur wenn angekreuzt.

mechtelm commented:
Zum Wiederherstellen von Werten in If-Bereichen: Habe auf Anhieb kein System dahinter entdeckt. Es zeigt sich, wenn man im Testbed mehrfach vor- und zurückblättert, dass die Werte schrittweise verschwinden. Das scheint auch unabhängig davon, ob der If-Bereich in einem Repeat-Block liegt oder nicht.

rhenck committed:
Change nav button icons to more fitting ones

rhenck commented:
> Zu if bei Checkboxes: Probier mal: Zum Start ist wird weder der true- noch der false-Zweig gezeigt, nach dem Klick nur der else-Zweig, und nur wenn angekreuzt.

Ja genau, mir ist klar, wie es gerade funktioniert. Ich dachte das ist vielleicht so beabsichtigt. Weil es halt so war, als ich das von dir übernommen habe.
Also nochmal konkret: Der Wunsch ist, dass im Urzustand der Else-Zweig aktiv ist?

mechtelm commented:
Ah weil bei If-Bereichen, die sich aus MC (Radios, Dropdown) ergeben, kein Wert vorbelegt ist. Also dann JA: Nach Start ist eine Checkbox false, d. h. auch der False-Zweig des If-Bereiches muss aktiv sein.

Aber das Problem war eigentlich, dass selbst beim Anklicken die If-Auswertung falsch war.

rhenck committed:
Make check box default value 'false'

This makes it so, that the false branch on depending If-Blocks is shown 
by default. Also all checkbox values are part of the result variable 
list, even without the user touching anything.

rhenck committed:
Merge branch 'master' into nav-buttons
